contracts/Dixel.sol

```diff
@@ -109,24 +111,26 @@ contract Dixel is Ownable, ReentrancyGuard, DixelSVGGenerator {
         Player storage player = _getOrAddPlayer(msgSender);
 
         uint256 totalPrice = 0;
-        for (uint256 i = 0; i < params.length; i++) {
-            Pixel storage pixel = pixels[params[i].x][params[i].y];
-            uint200 oldPrice = pixel.price;
+        unchecked {
+            for (uint256 i = 0; i < params.length; i++) {
+                Pixel storage pixel = pixels[params[i].x][params[i].y];
+                uint200 oldPrice = pixel.price;
 
-            pixel.color = params[i].color;
-            pixel.owner = player.id;
-            totalPrice += oldPrice;
+                pixel.color = params[i].color;
+                pixel.owner = player.id;
+                totalPrice += oldPrice;
 
-            pixel.price = uint200(oldPrice + oldPrice * PRICE_INCREASE_RATE / MAX_RATE);
-            require(pixel.price > oldPrice, "MAX_PRICE_REACHED");
+                pixel.price = uint200(oldPrice + oldPrice * PRICE_INCREASE_RATE / MAX_RATE);
+                require(pixel.price > oldPrice, "MAX_PRICE_REACHED");
+            }
         }
 
         // 10% goes to the contributor reward pools
-        uint256 reward = totalPrice * REWARD_RATE / MAX_RATE;
-        require(baseToken.transferFrom(msgSender, address(this), reward), "REWARD_TRANSFER_FAILED");
+        uint256 reward = (totalPrice * REWARD_RATE) / MAX_RATE;
+        assert(safeTransferFrom(baseToken, msgSender, address(this), reward));
```

Comment:
@sydneyitguy @Nipol  how this is a gas-saver improvement? seeing for example here https://medium.com/layerx/how-to-reduce-gas-cost-in-solidity-f2e5321e0395#2717, `assert` is worse than `require`

just trying to improve my skills, and understand better

Reply:
Oh I will try replacing all asset with require and check which one consumes more gas.
Thanks for pointing this out!

Reply:
Ah! Guys @inmarelibero @sydneyitguy 
In fact, it should have been separated into several PRs, but there is a part that internalized external calls through its own `Safe~~`. 
In addition, most contexts are not dynamic information, and errors during execution are only unique cases. And in all code dynamic verification tools with `assert` works optimally!

Reply:
This change is more optimized for code verification than gas.